packages/node/src/executors/execute/execute.impl.ts

```diff
@@ -57,68 +71,66 @@ export async function* executeExecutor(
     }
   }
 
-  for await (const event of startBuild(options, context)) {
-    if (!event.success) {
-      logger.error('There was an error with the build. See above.');
-      logger.info(`${event.outfile} was not restarted.`);
+  for await (const event of scheduleBuildAndProcessEvents(
+    startBuild(options, context),
+    processRunner
+  )) {
+    if (event.type === 'build') {
+      if (!event.value.success) {
+        logger.error('There was an error with the build. See above.');
+        logger.info(`${event.value.outfile} was not restarted.`);
+      }
     }
-    await handleBuildEvent(event, options);
-    yield event;
-  }
-}
 
-function runProcess(event: NodeBuildEvent, options: NodeExecuteBuilderOptions) {
-  if (subProcess || !event.success) {
-    return;
+    if (event.type === 'build' || options.emitSubprocessEvents) {
+      yield event.value;
+    }
   }
-
-  subProcess = fork(event.outfile, options.args, {
-    execArgv: getExecArgv(options),
-  });
 }
 
-function getExecArgv(options: NodeExecuteBuilderOptions) {
-  const args = ['-r', 'source-map-support/register', ...options.runtimeArgs];
-
-  if (options.inspect === true) {
-    options.inspect = InspectType.Inspect;
-  }
-
-  if (options.inspect) {
-    args.push(`--${options.inspect}=${options.host}:${options.port}`);
-  }
-
-  return args;
-}
+const nextIterValue = (typedIter: TypedIterator) =>
+  typedIter.type === 'build'
+    ? typedIter.iterator.next().then((result) => ({
+        iterator: typedIter.iterator,
+        type: typedIter.type,
+        ...result,
+      }))
+    : typedIter.iterator.next().then((result) => ({
+        iterator: typedIter.iterator,
+        type: typedIter.type,
+        ...result,
+      }));
+
+async function* scheduleBuildAndProcessEvents(
+  buildIterator: AsyncIterator<NodeBuildEvent>,
+  subProcessRunner: SubProcessRunner
+): AsyncGenerator<ScheduledEvent> {
+  const iteratorMap = new Map<
+    AsyncIterator<SubProcessEvent | NodeBuildEvent>,
+    ReturnType<typeof nextIterValue>
+  >([
+    [buildIterator, nextIterValue({ type: 'build', iterator: buildIterator })],
+  ]);
+
+  while (iteratorMap.size > 0) {
+    const result = await Promise.race(iteratorMap.values());
```

Comment:
Greedily take the next ready value. 

Consume each individual iterator lazily: Only tick the last iterator to have won the race.
Remove complete iterators from the race.